src/suunto_vyper_parser.c

```diff
@@ -297,6 +304,17 @@ suunto_vyper_parser_get_field (dc_parser_t *abstract, dc_field_type_t type, unsi
 				*((dc_divemode_t *) value) = DC_DIVEMODE_OC;
 			}
 			break;
+		case DC_FIELD_STRING:
+			switch(flags) {
+			case 0: /* serial */
+				string->desc = "Serial";
+				snprintf(buf, BUFLEN, "%u", parser->serial);
```

Comment:
Hmm. I think Suunto serial numbers are actually fixed-length, so they _could_ start with a zero, and this will drop them.

I doubt it matters in practice, and they have some meaning that indicates manufacturing year and month, so it is possible that zero never happens.

I'll merge this as-is, but it's a classic case of "printing it as a number may lose information".

Reply:
I looked at that trying to find instances of people talking about serial numbers for these older dive computers... I found a hand full of references - and none started with a zero... so... 🤷🏼‍♂️